sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java

```diff
@@ -28,47 +33,131 @@ public NetworkCallError() {
-                return new IndexOutOfBoundsException(throwable.getMessage());
+    /**
+     * Gets the name of the class of the throwable. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @return The class name which is Class type.
+     */
+    public Class<?> className() {
+        return this.className;
+    }
 
-            case "java.net.UnknownHostException":
-                return new UnknownHostException(throwable.getMessage());
+    /**
+     * Sets the throwable argument types which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @param argTypes The array of argument types.
+     * @return The instance of NetworkCallError class.
+     */
+    public NetworkCallError argTypes(Class<?>[] argTypes) {
+        this.argTypes = ImplUtils.clone(argTypes);
+        return this;
+    }
 
-            default:
-                return throwable;
-        }
+    /**
+     * Gets the throwable argument types which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @return The array of argument types.
+     */
+    public Class<?>[] argTypes() {
+        return ImplUtils.clone(this.argTypes);
     }
 
     /**
-     * Sets the throwable that was thrown during a network call.
+     * Sets the throwable argument default values which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
      *
-     * @param throwable Throwable that was thrown.
+     * @param argValues The array of argument values.
+     * @return The instance of NetworkCallError class.
      */
-    public void throwable(Throwable throwable) {
-        this.throwable = throwable;
+    public NetworkCallError argValues(Object[] argValues) {
+        this.argValues = ImplUtils.clone(argValues);
+        return this;
     }
 
     /**
-     * Sets the name of the class of the throwable. This is used during deserialization the construct the throwable
+     * Gets the throwable argument default values which needed for new instance constructor. This is used during deserialization the construct the throwable
      * as the actual class that was thrown.
      *
-     * @param className Class name of the throwable.
+     * @return The array of argument values.
      */
-    public void className(String className) {
-        this.className = className;
+    public Object[] argValues() {
+        return ImplUtils.clone(this.argValues);
+    }
+
+    private Object getDefaultValue(Class<?> type) throws Exception {
+        // Get default value of non-primitive type
+        if (!type.isPrimitive()) {
+            return type.getConstructor().newInstance();
```

Comment:
Is there always a guarantee for an empty constructor?

Reply:
The record did not work for me. The live tests also failed at `AuthenticationMismatchError`
I will create an issue and link it in code. Issue opened  https://github.com/Azure/azure-sdk-for-java/issues/5172